test/e2e/serial/tests/workload_unschedulable.go

```diff
@@ -229,4 +231,138 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload unsched
+			}
+		})
+		It("[test_id:47615][tier2][unsched] a deployment with multiple guaranteed pods resources that doesn't fit at the NUMA level", func() {
+			By("padding all the nodes")
+			requiredRes := corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("4"),
+				corev1.ResourceMemory: resource.MustParse("4G"),
+			}
+
+			padUntilRes := corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("3"),
+				corev1.ResourceMemory: resource.MustParse("8G"),
+			}
+			err := padder.Nodes(len(nrts)).UntilAvailableIsResourceList(padUntilRes).Pad(time.Minute*2, e2epadder.PaddingOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			nrtInitialList, err := e2enrt.GetUpdated(fxt.Client, nrtv1alpha1.NodeResourceTopologyList{}, time.Second*10)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("creating a deployment")
+			dpName := "test-dp"
+			schedulerName := nrosched.NROSchedulerName
+			replicas := int32(6)
+			podLabels := map[string]string{
+				"test": "test-dp",
+			}
+
+			podSpec := corev1.PodSpec{
+				SchedulerName: schedulerName,
+				Containers: []corev1.Container{
+					{
+						Name:    dpName + "-cnt1",
+						Image:   objects.PauseImage,
+						Command: []string{objects.PauseCommand},
+						Resources: corev1.ResourceRequirements{
+							Requests: requiredRes,
+							Limits:   requiredRes,
+						},
+					},
+				},
+				RestartPolicy: corev1.RestartPolicyAlways,
+			}
+			dp := objects.NewTestDeploymentWithPodSpec(replicas, podLabels, map[string]string{}, fxt.Namespace.Name, dpName, podSpec)
+
+			err = fxt.Client.Create(context.TODO(), dp)
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("checking deployment pods have been scheduled with the topology aware scheduler %q ", schedulerName))
+			pods, err := schedutils.ListPodsByDeployment(fxt.Client, *dp)
+			Expect(err).ToNot(HaveOccurred(), "unable to get pods from deployment %q:  %v", dp.Name, err)
+
+			allFailed := true
+			for _, pod := range pods {
+				isFailed, err := nrosched.CheckPODSchedulingFailed(fxt.K8sClient, pod.Namespace, pod.Name, schedulerName)
+				if err != nil {
+					_ = objects.LogEventsForPod(fxt.K8sClient, pod.Namespace, pod.Name)
+				}
+				Expect(err).ToNot(HaveOccurred())
+				allFailed = allFailed && isFailed
+				if !isFailed {
+					klog.Warningf("pod %s/%s with scheduler %s did NOT fail", pod.Namespace, pod.Name, schedulerName)
+					continue
+				}
+			}
+			Expect(allFailed).To(BeTrue(), "some pods are running, but we expect all of them to fail")
+
+			By("updating deployment in such way that some pods will fit into NUMA nodes")
+			err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(dp), dp)
+			Expect(err).ToNot(HaveOccurred())
+
+			// 6 pods in total (replica is 6)
+			// we should expect 'expectedReadyReplicas' out of 6 pods to be running
+			expectedReadyReplicas := calcExpectedReadyReplicas(numOfnrtCandidates, len(nrts)-numOfnrtCandidates)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
this seems stale. Also, we probably should `klog.Infof` the number of expected replicas

Reply:
What do you mean by stale? this is a new code

Reply:
I mean that on line 315 you check for the error not to have occurred (fine). Then we don't write to this value again, and here above we run the same check which is pointless because we didn't write to this value again. Hence this check seems stale (copy/paste mistake?)